packages/samples/headless-ssr/src/components/result-list.tsx

```diff
@@ -1,18 +1,39 @@
-'use client';
+import {
+  ResultList as ResultListController,
+  ResultListState,
+} from '@coveo/headless';
+import {useEffect, useState, FunctionComponent} from 'react';
 
-import {Result} from '@coveo/headless';
+interface ResultListProps {
+  initialState: ResultListState;
+  controller?: ResultListController;
+}
+
+export const ResultList: FunctionComponent<ResultListProps> = (props) => {
+  const {initialState, controller} = props;
+  const [state, setState] = useState(initialState);
+
+  useEffect(
+    () => controller?.subscribe(() => setState({...controller.state})),
+    [controller]
+  );
+
+  if (!state.results.length) {
+    return <div>No results</div>;
+  }
 
-export default function ResultList({results}: {results: Result[]}) {
   return (
     <div>
```

Comment:
The e2e tests check for number of results etc - so they needed to be added back.
I also copied the initial draft from the existing result-list.fn.tsx sample in headless-react-samples but then modified it to simplify the markup to the bare essentials necessary for illustration. 
Added the `clickUri` as an example of an interactive element.
If we were able to import the existing markup and reuse it - yes there would be advantages to keep it the same. But I do not see any advantage in trying to align the markup with the existing headless-react-samples.